include/identity_walker.h

```diff
@@ -44,36 +44,66 @@ enum WalkerStepResult
  * If clear_cache is set, the cache is cleared between calls to visit
  * otherwise, the cache persists
  *
+ * The user can optionally pass a pointer to a cache. If that pointer
+ * is non-null, it will be used in place of the internal cache.
+ *
  * Important Note: The term arguments should belong to the solver provided
  * to the identity walker, otherwise the behavior is undefined.
  */
 class IdentityWalker
 {
 public:
-  IdentityWalker(smt::SmtSolver & solver, bool clear_cache)
-    : solver_(solver), clear_cache_(clear_cache) {};
+ IdentityWalker(smt::SmtSolver & solver,
+                bool clear_cache,
+                smt::UnorderedTermMap * ext_cache = nullptr)
+     : solver_(solver), clear_cache_(clear_cache), ext_cache_(ext_cache){};
 
-  /** Visit a term and all its subterms in a pre- and post-order traversal
-   *  @param term the term to visit
-   *  @return the term after visiting (returns the value of cache[term]
-   *     -- if it has been cached and returns term otherwise)
-   */
-  smt::Term visit(smt::Term & term);
+ /** Visit a term and all its subterms in a pre- and post-order traversal
```

Comment:
pre- and post-order ? Isn't it one?

Reply:
Updated the comment with b1ff8dd5035c096443dab83b266eaf12c25d925a. Is that clearer?

Reply:
yup :)